### PR TITLE
Tidy the build files

### PR DESCRIPTION
**Pull request by 3flex**
Just for fun I went through the build files to look for possible improvements - this should make them a little bit clearer and performant.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -77,9 +78,7 @@ allprojects {
     version = detektVersion + if (System.getProperty("snapshot")?.toBoolean() == true) "-SNAPSHOT" else ""
 
     repositories {
-        mavenLocal()
         jcenter()
-        maven(url = "https://dl.bintray.com/arturbosch/generic")
```
Comment (schalkms) [inline build.gradle.kts]: Why did you remove both `maven` instructions? Is this not necessary anymore?

Comment (3flex) [reply build.gradle.kts]: The build won't succeed if repositories are required that are removed. The bintray line was there because there used to be a dependency on ksh library (from memory) which is no longer used.

mavenLocal would have supported local testing of the Gradle plugin but with composite builds that's not necessary anymore.

Currently not all dependencies are available on Maven Central otherwise would use that instead of jcenter.

File: build.gradle.kts [hunk 2]
```diff
@@ -139,7 +138,7 @@ subprojects {
         }
     }
 
-    tasks.withType<Test> {
+    tasks.withType<Test>().configureEach {
```
Comment (arturbosch) [inline build.gradle.kts]: This does lazy configuration?

File: build.gradle.kts [hunk 3]
```diff
@@ -242,32 +231,32 @@ subprojects {
...
-                    license.appendNode("url", "http://www.apache.org/licenses/LICENSE-2.0.txt")
-                    license.appendNode("distribution", "repo")
-
-                    val developer = appendNode("developers").appendNode("developer")
-                    developer.appendNode("id", "Artur Bosch")
-                    developer.appendNode("name", "Artur Bosch")
-                    developer.appendNode("email", "arturbosch@gmx.de")
-
-                    appendNode("scm").appendNode("url", "https://github.com/arturbosch/detekt")
+            pom {
```
Comment (arturbosch) [inline build.gradle.kts]: Much nicer :)

File: build.gradle.kts (hunk 2 above)
Comment (3flex) [reply build.gradle.kts]: Lazy configuration and task configuration avoidance are two different things. This is task configuration avoidance (tasks are configured on demand).

https://docs.gradle.org/current/userguide/task_configuration_avoidance.html
